src/test/java/seedu/address/testutil/PersonBuilder.java

```diff
@@ -115,6 +116,14 @@ public PersonBuilder withEmail(String email) {
         return this;
     }
 
+    /**
+     * Sets the {@code Tag} of the {@code Person} that we are building.
+     */
+    public PersonBuilder withTag(String tagName, TagStatus tagStatus) {
+        this.tags = Tag.updateTagsWithNewTag(new HashSet<>(tags), tagName, tagStatus);
+        return this;
+    }
+
```

Comment:
The current name is a bit confusing, can change to something like "updateTag'.
Also it is creating a hashset every time so a bit inefficient (but can ignore for now).